### ISSUE TITLE
velero pod on source side should not go to CrashLoopBackoff upon access permissions to bucket

### ISSUE THREAD

**Issue by eriknelson** [konveyor/mig-controller#447]
Apologies, I can't seem to open issues against our velero fork and there isn't a ton to go off of here, but maybe it rings a bell for someone:

> (BUG) velero pod on source side should not go to CrashLoopBackoff upon access permissions to bucket
See doc for log output

```
An error occurred: some backup storage locations are invalid: backup store for location "repo-large1-szbjp" is invalid: rpc error: code = Unknown desc = AccessDenied: Access Denied
    status code: 403, request id: k73di8h0-c12w4a-156n, host id: k73di8h0-c12w4a-156n
```

This is about as much as I've got. Feel free to close if this is not sufficient or this belongs somewhere else, just trying to catalog issues that came up during the hackathon.

**Comment by dymurray** [comment]
This looks like it would be a valid error if the object storage was the only BSL configured and had invalid credentials. We should detect this prior to creating the BSL so that we don't put Velero in this state.

This is worth investigating.

**Comment by dymurray** [comment]
https://github.com/vmware-tanzu/velero/issues/1967

**Comment by sseago** [comment]
I think I ran into this same problem a couple weeks ago. I had two BSLs but one of them became invalid (I think the NooBaa server crashed), and velero refused to start until I deleted the NooBaa BSL.

**Comment by carlisia** [comment]
Yes, this is currently the way Velero behaves, ie, if it finds an invalid BSL, it will block operations. We are changing the behavior so if that Velero continues creating backups even if there are invalid BSLs, as long as there's a valid BSL for them. Work being tracked here: https://github.com/vmware-tanzu/velero/issues/1967

**Comment by djwhatle** [comment]
@dymurray @sseago did this ever get fixed in Velero upstream? Wondering if we should keep the issue open.

**Comment by sseago** [comment]
It looks like the referenced PR was merged post-1.4.2, and it's been refactored since then. I'm guessing that 1.5.2 has the fix in one form or another, but I'm not 100% sure.

**Comment by djwhatle** [comment]
Ok, will leave this open for now.
